src/services/message.service.ts

```diff
@@ -31,6 +31,57 @@ export class MessageService {
     this._sendConstructedReport(report, { files: message.attachments.map((e) => e.url) });
   }
 
+  async sendTextMessage(
+    message: IMessage,
+    content: string,
+    argOptions: ITextMessageOptions
+  ): Promise<void> {
+    const options: ITextMessageOptions = {};
+    options.header = argOptions.header ? `${argOptions.header}\n` : '';
+    options.footer = argOptions.footer ? `\n${argOptions.footer}` : '';
+    options.reply = argOptions.reply || false;
+    options.delimiter = argOptions.delimiter === false ? false : argOptions.delimiter || '\n';
+
+    const replyLength = options.reply ? `${message.author},  `.length : 0;
+    const templateLength = replyLength + options.header.length + options.footer.length;
+
+    let messagesToSend: string[] | null = [`${options.header}${content}${options.footer}`];
+
+    if (content.length + templateLength > Constants.MaxMessageLength) {
+      if (!options.delimiter) {
+        return Promise.reject(`Message was over Discords character cap.`);
+      }
+
+      options.delimiter = argOptions.delimiter === '*' ? '' : options.delimiter;
+
+      messagesToSend = this._constructMessages(
+        content.split(options.delimiter),
+        templateLength,
+        options.delimiter,
+        options.header,
+        options.footer
+      );
+
+      messagesToSend.forEach((data) => {
+        if (data.length > Constants.MaxMessageLength) {
+          return Promise.reject(`Unable to split the message up this much.`);
+        }
+      });
```

Comment:
What's the point of the `Promise.reject` in here? Plus it does nothing because all it does is return out of the `forEach`

Reply:
idea was to catch if any messages from a set failed to send. requested [here](https://github.com/cs-discord-at-ucf/lion/pull/336#discussion_r604184280)  Honestly not sure how you would check to see all promises were fulfilled.

Reply:
The return just returns out of the forEach, so it doesn't do anything though